pandas/_libs/groupby_helper.pxi.in

```diff
@@ -571,17 +575,22 @@ def group_rank_{{name}}(ndarray[float64_t, ndim=2] out,
             # (used by pct calculations later). also be sure to reset any of
             # the items helping to calculate dups
             if i == N - 1 or labels[_as[i]] != labels[_as[i+1]]:
-                for j in range(grp_start, i + 1):
-                    grp_sizes[_as[j], 0] = i - grp_start + 1 - grp_na_count
-                dups = sum_ranks = 0
+                if pct:
+                    if tiebreak != TIEBREAK_DENSE:
```

Comment:
Instead of doing this here is it not possible to alter the handling of the `tie_count` variable incrementing above depending on whether or not we are using `pct` and `TIEBREAK_DENSE` together? Seems like it would be simpler to do that then to implement another branch for assigning percents, if that's possible

Reply:
Do you mean move the logic to here ?
```
                elif tiebreak == TIEBREAK_DENSE:
                    for j in range(i - dups + 1, i + 1):
                        out[_as[j], 0] = grp_vals_seen
```

Reply:
Hmm well I suppose that that would alter the non-pct items, but my overall point is I feel like this can be done more succinctly. In either case (TIEBREAK_DENSE or not) it's a matter of dividing by the right denominator to get the correct value. 

So I think it would be cleaner to do something along the lines of:

```python
if TIEBREAK_DENSE:
    denominator = ...
else:
    denominator = ...

for j in range(lab_start, i + 1):
    out[_as[j], 0] = out[_as[j], 0] / denominator
```